scripts/lib/CIME/SystemTests/system_tests_compare_two.py

```diff
@@ -173,33 +173,37 @@ def _case_two_custom_postrun_action(self):
     # ========================================================================
 
     def build_phase(self, sharedlib_only=False, model_only=False):
-        if self._separate_builds:
-            self._activate_case1()
-            self.build_indv(sharedlib_only=sharedlib_only, model_only=model_only)
-            self._activate_case2()
-            # Although we're doing separate builds, it still makes sense
-            # to share the sharedlibroot area with case1 so we can reuse
-            # pieces of the build from there.
-            if get_model() != "e3sm":
-                # We need to turn off this change for E3SM because it breaks
-                # the MPAS build system
-                self._case2.set_value("SHAREDLIBROOT",
-                                      self._case1.get_value("SHAREDLIBROOT"))
-
-            self.build_indv(sharedlib_only=sharedlib_only, model_only=model_only)
-        else:
-            self._activate_case1()
-            self.build_indv(sharedlib_only=sharedlib_only, model_only=model_only)
-            # pio_typename may be changed during the build if the default is not a
-            # valid value for this build, update case2 to reflect this change
-            for comp in self._case1.get_values("COMP_CLASSES"):
-                comp_pio_typename = "{}_PIO_TYPENAME".format(comp)
-                self._case2.set_value(comp_pio_typename, self._case1.get_value(comp_pio_typename))
-
-            # The following is needed when _case_two_setup has a case_setup call
-            # despite sharing the build (e.g., to change NTHRDS)
-            self._case2.set_value("BUILD_COMPLETE",True)
-            self._case2.flush()
+        # Subtle issue: case1 is already in a writeable state since it tends to be opened
+        # with a with statement in all the API entrances in CIME. case2 was created via clone,
+        # not a with statement, so it's not in a writeable state, so we need to use a with
+        # statement here to put it in a writeable state.
+        with self._case2:
```

Comment:
I would have a slight preference to moving this `with` statement down, so that it just surrounds the code that actually needs it, even if that means we need two or three separate `with` blocks in this function. For example, I'd think that the `build_indv` call on case1 doesn't need to be within a `with self._case2` block. For those of us who don't fully understand where this `with` statement is needed, I think that would help us in reading / modifying the code later - i.e., if only the only things in `with` blocks are things that _need_ to be in `with` blocks.

But I don't feel strongly about this, so if you prefer it out at this level, that's okay with me.